### PR TITLE
[move modules] Add consensus key validation

### PR DESCRIPTION
**Pull request by valerini**
This PR adds validation of a new consensus key in ValidatorConfig move module.

### CODE REVIEW

File: language/stdlib/modules/ValidatorConfig.move [hunk 1]
```diff
@@ -88,6 +90,7 @@ module ValidatorConfig {
             Signer::address_of(signer) == get_operator(validator_account),
             1101
         );
+        Transaction::assert(Signature::ed25519_validate_pubkey(copy consensus_pubkey), 1108);
         // TODO(valerini): verify the validity of new_config.consensus_pubkey and
         // the proof of posession
```
Comment (huitseeker) [inline language/stdlib/modules/ValidatorConfig.move:95]: nit: see comment below

File: language/stdlib/modules/doc/ValidatorConfig.md [hunk 2]
```diff
@@ -230,6 +230,7 @@
         <a href="Signer.md#0x0_Signer_address_of">Signer::address_of</a>(signer) == <a href="#0x0_ValidatorConfig_get_operator">get_operator</a>(validator_account),
         1101
     );
+    Transaction::assert(<a href="Signature.md#0x0_Signature_ed25519_validate_pubkey">Signature::ed25519_validate_pubkey</a>(<b>copy</b> consensus_pubkey), 1108);
     // TODO(valerini): verify the validity of new_config.consensus_pubkey and
     // the proof of posession
```
Comment (huitseeker) [inline language/stdlib/modules/doc/ValidatorConfig.md:235]: nit: you may want to edit this comment, and possibly link it to a new GH issue if you don't plan to tackle posession right away

Comment (valerini) [reply language/stdlib/modules/doc/ValidatorConfig.md:235]: I actually wanted to tackle possession in this pull-request as well...

Comment (valerini) [reply language/stdlib/modules/doc/ValidatorConfig.md:235]: The proof-of-possession sparkled some interesting discussions, we'll tackle it later.

**Pull request closed without merging**